pages/instructorAssessmentInstances/instructorAssessmentInstances.ejs

```diff
@@ -93,10 +173,26 @@
         <div class="card-header bg-primary text-white d-flex align-items-center">
           <%= assessment_set.name %> <%= assessment.number %>: Assessment instances
           <div class="ml-auto">
-            <button type="button" class="btn btn-sm btn-light" tabindex="0"
-               data-toggle="modal" data-target="#deleteAllAssessmentInstancesModal">
-              <i class="fa fa-times" aria-hidden="true"></i> Delete all assessment instances
-            </button>
+            <div class="dropdown d-flex flex-row">
+              <button type="button" class="btn btn-light dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
+                Action for all instances <span class="caret"></span>
+              </button>
+              <!-- Capture all clicks to dropdown items to prevent scrolling to the top of the page -->
+              <div class="dropdown-menu dropdown-menu-right" onclick="window.event.preventDefault();">
+                
+                <% if (authz_data.has_instructor_edit) { %>
+                <a class="dropdown-item"
+                   href="#"
+                   data-toggle="modal" data-target="#deleteAllAssessmentInstancesModal">
+                  <i class="fa fa-times" aria-hidden="true"></i> Delete all instances
+                </a>
+                <a class="dropdown-item time-limit-edit-button" data-row='<%= JSON.stringify(time_limit_totals) %>'>
```

Comment:
Nit: single quotes aren't necessary since EJS will escape the JSON string for you.

```suggestion
                <a class="dropdown-item time-limit-edit-button" data-row="<%= JSON.stringify(time_limit_totals) %>">
```